src/components/views/rooms/MemberList.js

```diff
@@ -325,27 +348,84 @@ module.exports = React.createClass({
+            // }
+
+            if (!userA && !userB) return 0;
+            if (userA && !userB) return -1;
+            if (!userA && userB) return 1;
+
+            // console.log("comparing " + this.memberString(memberA) + " and " + this.memberString(memberB));
+
+            if (userA.currentlyActive && userB.currentlyActive) {
+                // console.log(memberA.name + " and " + memberB.name + " are both active");
+                if (memberA.powerLevel === memberB.powerLevel) {
+                    // console.log(memberA + " and " + memberB + " have same power level");
+                    if (memberA.name && memberB.name) {
+                        // console.log("comparing names: " + memberA.name + " and " + memberB.name);
+                        return memberA.name.localeCompare(memberB.name);
+                    }
+                    else {
+                        return 0;
+                    }
+                }
+                else {
+                    // console.log("comparing power: " + memberA.powerLevel + " and " + memberB.powerLevel);
+                    return memberB.powerLevel - memberA.powerLevel;
+                }
+            }
 
-        var presenceMap = {
-            online: 3,
-            unavailable: 2,
-            offline: 1
-        };
+            if (userA.currentlyActive && !userB.currentlyActive) return -1;
+            if (!userA.currentlyActive && userB.currentlyActive) return 1;
 
-        var presenceOrdA = userA ? presenceMap[userA.presence] : 0;
-        var presenceOrdB = userB ? presenceMap[userB.presence] : 0;
+            // For now, let's just order things by timestamp. It's really annoying
+            // that a user disappears from sight just because they temporarily go offline
+    /*
+            var presenceMap = {
+                online: 3,
+                unavailable: 2,
+                offline: 1
+            };
 
-        if (presenceOrdA != presenceOrdB) {
-            return presenceOrdB - presenceOrdA;
-        }
+            var presenceOrdA = userA ? presenceMap[userA.presence] : 0;
+            var presenceOrdB = userB ? presenceMap[userB.presence] : 0;
 
-        var lastActiveTsA = userA && userA.lastActiveAgo ? userA.lastPresenceTs - userA.lastActiveAgo : 0;
-        var lastActiveTsB = userB && userB.lastActiveAgo ? userB.lastPresenceTs - userB.lastActiveAgo : 0;
+            if (presenceOrdA != presenceOrdB) {
+                return presenceOrdB - presenceOrdA;
+            }
+    */
 
-        return lastActiveTsB - lastActiveTsA;
+            var lastActiveTsA = userA && userA.lastActiveAgo ? (userA.lastPresenceTs - userA.lastActiveAgo) : Number.MAX_SAFE_INTEGER;
+            var lastActiveTsB = userB && userB.lastActiveAgo ? (userB.lastPresenceTs - userB.lastActiveAgo) : Number.MAX_SAFE_INTEGER;
+
+            // console.log("comparing ts: " + lastActiveTsA + " and " + lastActiveTsB);
+
+            return lastActiveTsA - lastActiveTsB;
```

Comment:
I think this is now wrong? We want most recently active first, which means we want highest `lastActiveTs` first, which means we should return < 0 if A has a higher `lastActiveTs` than B.